### DISCUSSION TITLE
Retrying task on requests.on_failure() o WorkerLostError

### DISCUSSION THREAD

**Discussion by Santifire92** [celery/celery#9210] [Q&A]
I'm currently handling retrying tasks on a WorkerLostError.
I know i can wrap the Requests with a custom on_failure method but i don't know how shoul di call the task within the request to get rerunned.
`class MyRequest(Request):

    def on_failure(self, exc_info, send_failed_event=True, return_ok=False):
        exc = exc_info.exception
        self.retries = getattr(self, 'retries', 0)
        if isinstance(exc, ExceptionWithTraceback):
            exc = exc.exc
            if isinstance(exc, WorkerLostError):
                logger.error('Worker lost error, retrying task')
                if self.retries < 2:
                    self.retries += 1
                    return self.execute()
        super().on_failure(
            exc_info=exc_info,
            send_failed_event=send_failed_event,
            return_ok=return_ok
        )`
I know it's horrible but i want to get it working first and know which is the correct way of signaling the task to get rerunned, given that the logic of task.retries doesn't work on WorkerLostErrors situations

**Comment by bashir-abdelwahed** [comment]
I have the same issue. I need to retry on worker lost. I think what you are doing is perfectly fine because it is impossible to call task.retry method from inside the worker that receives sig9 (no handler allowed for this signal); hence the retry must be done in the main process.

**Comment by bashir-abdelwahed** [comment]
However, in your current implementation, the request is re-executed in the main process instead of being executed in a child process which is dangerous.

**Comment by bashir-abdelwahed** [comment]
I managed to make the request retry by making it reque, but it is such a pain to be honest.
I think we need some help from a core developer of celery
